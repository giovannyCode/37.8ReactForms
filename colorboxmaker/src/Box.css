.box {
  position: relative;
  display: inline-block;
}

.close-btn {
  position: absolute;
  top: 5px;
  right: 10px;
  cursor: pointer;
  font-size: 20px;
  color: black;
  font-weight: bold;
}

.close-btn:hover {
  color: red;
}
