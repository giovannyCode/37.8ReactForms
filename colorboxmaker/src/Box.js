import React from "react";
import "./Box.css";

function Box({ color, width, height, deleteBox, id }) {
  return (
    <div
      className="box"
      style={{ backgroundColor: color, width: width, height: height }}
    >
      <button onClick={() => deleteBox(id)} className="close-btn">
        X
      </button>
    </div>
  );
}

export default Box;
