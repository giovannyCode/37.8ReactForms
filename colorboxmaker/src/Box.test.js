import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function () {
  const mockDeleteBox = jest.fn();
  render(
    <Box
      key="1"
      width="150px"
      height="200px"
      color="Cian"
      deleteBox={mockDeleteBox}
      id="1"
    />
  );
});

it("matches snapshot", function () {
  const mockDeleteBox = jest.fn();
  const { asFragment } = render(
    <Box
      key="1"
      width="150px"
      height="200px"
      color="Cian"
      deleteBox={mockDeleteBox}
      id="1"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

/* 
it("works when you click on the right arrow", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("expects that when you’re on the second image, clicking the left arrow will move you to the first image", function () {
  const { container, debug } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  fireEvent.click(leftArrow);
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
});

it("checks that the left arrow is missing when you’re on the first image", function () {
  const { container } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );

  expect(
    container.querySelector(".bi.bi-arrow-left-circle")
  ).not.toBeInTheDocument();
});

it("checks that the rigth arrow is missing when you’re on the third image", function () {
  const { container, debug } = render(
    <Carousel photos={TEST_IMAGES} title="images for testing" />
  );
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);
  expect(
    container.querySelector(".bi.bi-arrow-right-circle")
  ).not.toBeInTheDocument();
});
 */
