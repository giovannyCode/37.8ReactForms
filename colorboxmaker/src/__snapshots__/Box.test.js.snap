// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
    style="width: 150px; height: 200px;"
  >
    <button
      class="close-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
